api/bag_private.h

```diff
@@ -1,192 +1,113 @@
-bagError bagWriteAttribute  (bagHandle hnd, hid_t lid, u8 *attr_name, void *value);
-bagError bagReadAttribute   (bagHandle hnd, hid_t lid, u8 *attr_name, void *value);
-bagError bagAlignAttribute  (bagHandle hnd, hid_t lid, u8 *attr_name, void *value, u32 read_or_write);
-bagError bagReadSurfaceDims (bagHandle hnd, hsize_t *max_dims);
-bagError bagReadOptSurfaceDims (bagHandle hnd, s32 type);
-bagError bagAlignXMLStream  (bagHandle hnd, s32 read_or_write);
-bagError bagAlignRow        (bagHandle hnd, u32 row, u32 start_col,u32 end_col, s32 type, s32 read_or_write, void *data);
-bagError bagAlignRegion     (bagHandle hnd, u32 start_row, u32 start_col, u32 end_row, u32 end_col, s32 type, s32 read_or_write, hid_t xfer);
-bagError bagAlignNode       (bagHandle hnd, u32 row, u32 col, s32 type, void *data, s32 read_or_write);
-bagError bagAlignOptRow     (bagHandle hnd, u32 row, u32 start_col,u32 end_col, s32 type, s32 read_or_write, void *data);
-bagError bagAlignOptRegion  (bagHandle hnd, u32 start_row, u32 start_col, u32 end_row, u32 end_col, s32 type, s32 read_or_write, hid_t xfer);
-bagError bagAlignOptNode    (bagHandle hnd, u32 row, u32 col, s32 type, void *data, s32 read_or_write);
-bagError bagUpdateMinMax    (bagHandle hnd, u32 type);
-bagError bagReadTrackingList(bagHandle hnd, u16 mode, u32 inp1, u32 inp2, bagTrackingItem **items, u32 *rtn_len);
-bagError bagFillPos         (bagHandle hnd, u32 r1, u32 c1 , u32 r2, u32 c2, f64 **x, f64 **y);
-bagError bagSortTrackingList(bagHandle hnd, u16 mode);
-s32 bagCompareTrackIndices  (const void *a, const void *b);
-s32 bagCompareTrackNodes    (const void *a, const void *b);
-s32 bagCompareTrackCodes    (const void *a, const void *b);
-
-#endif
+namespace BAG
+{
+
+//! The maximum compression level supported by HDF5.
+constexpr int kMaxCompressionLevel = 9;
+
+//! Path names for BAG entities
+#define ROOT_PATH                       "/BAG_root"
+#define METADATA_PATH                   ROOT_PATH "/metadata"
+#define ELEVATION_PATH                  ROOT_PATH "/elevation"
+#define UNCERTAINTY_PATH                ROOT_PATH "/uncertainty"
+#define TRACKING_LIST_PATH              ROOT_PATH "/tracking_list"
+
+//! Path names for optional BAG entities
+#define NOMINAL_ELEVATION_PATH          ROOT_PATH "/nominal_elevation"
+#define VERT_DATUM_CORR_PATH            ROOT_PATH "/vertical_datum_corrections"
+#define HYPOTHESIS_STRENGTH_PATH        ROOT_PATH "/hypotheses_strength"
+#define NUM_HYPOTHESES_PATH             ROOT_PATH "/num_hypotheses"
+#define AVERAGE_PATH			        ROOT_PATH "/average"
+#define NODE_GROUP_PATH			        ROOT_PATH "/node"
+#define ELEVATION_SOLUTION_GROUP_PATH   ROOT_PATH "/elevation_solution"
+#define SHOAL_ELEVATION_PATH            ROOT_PATH "/shoal_elevation"
+#define STANDARD_DEV_PATH	            ROOT_PATH "/standard_dev"
+#define NUM_SOUNDINGS_PATH              ROOT_PATH "/num_soundings"
+#define COMPOUND_PATH                   ROOT_PATH "/Georef_metadata/"
```

Comment:
Why is "Georef_metadata" the only one with uppercase first letter?

Reply:
I'm not sure. @GlenRice-NOAA do you know?

Reply:
We can't change the structure of the BAG file, since GDAL has already implemented this, so I think we are stuck with /Georef_metadata. However, we can change `#define COMPOUND_PATH` to `#define GEOREF_META_PATH`, which will make the code easier to follow.

Reply:
We can update GDAL to read /georef_metadata.

Reply:
I would advise against this. Then we'd have to support both `/georef_metadata` and `/Georef_metadata`, otherwise, any existing files would be unreadable going forward. What I think we should change is how we refer to the georef metadata layer within the code. Currently, georef_metadata is referred to as a compound layer, which is confusing. Nothing else but georef_metadata layer makes use of this code, so it would seem safe to rename everything in terms of georef_metadata, replacing references to compound layer, etc. However, before doing this, I wanted to have the discussion of whether the "compound layer" name was chosen for a reason (e.g., to support anticipated future work).

Reply:
Since the only way to write a BAG with the metadata layer is this ONSWG BAG library, and the library only now is being updated to perform this function, I'm not too worried about offending anyone that would have a BAG with a `/Georef_metadata` group.  I think we should go for consistent.